cyclonedx-bom/src/models/bom.rs

```diff
@@ -563,15 +562,12 @@ pub enum UrnUuidError {
     InvalidUrnUuid(String),
 }
 
-fn matches_urn_uuid_regex(value: &str) -> Result<bool, regex::Error> {
-    static UUID_REGEX: Lazy<Result<Regex, regex::Error>> = Lazy::new(|| {
+fn matches_urn_uuid_regex(value: &str) -> bool {
```

Comment:
given that you had to change this code, it could be nice if `matches_urn_uuid_regex` returns a `ValidationResult` directly so `UrnUuid::new` and `UrnUuid::validate_with_context` don't have to map the `bool` to a `ValidationResult` themselves.

I wouldn't do this if `matches_urn_uuid_regex` is used elsewhere without mapping the `bool`.

Reply:
I think it's ok to keep it as is, function `matches_urn_uuid_regex` only checks if the given string matches, while the validation method maps that to the ValidationResult. The alternative could have been to merge both into one function.